### PR TITLE
Do not squeeze depth dim

### PR DESCRIPTION
**Pull request by jk-rieck**
#12 : I included `z_c`, `z_l` and `z` in the dimensions not to squeeze in `trim_and_squeeze()`. I also had to remove a `.squeeze()` in `load_xorca_dataset` (l 383) because that basically made `trim_and_squeeze()` ineffective for aux_files. I am not sure whether it has other effects...

### CODE REVIEW

File: xorca/lib.py [hunk 1]
```diff
@@ -46,6 +46,8 @@ def trim_and_squeeze(ds,
         "GLOBAL": {"y": (1, -1), "x": (1, -1)},
         "NEST": {},
     }
+    
+    z_dims = ["z_c", "z_l", "z"]
```
Comment (willirath) [inline xorca/lib.py]: Should we add `"deptht", "depthu", "depthv", "depthw"` here?

Also, I'd put this into `orca_names` and use a tuple because it's immutable?

Comment (jk-rieck) [reply xorca/lib.py]: I guess we do not need the `depth?`s as long as `rename_dims()` is called before `trim_and_squeeze()`... However, it also won't hurt to include them...

File: xorca/lib.py [hunk 2]
```diff
@@ -62,10 +62,10 @@ def trim_and_squeeze(ds,
         ds = ds.isel(x=slice(*x_slice))
 
     to_squeeze = [dim for dim in ds.dims
-                  if ((ds[dim].size == 1) and ((dim is not "t")
+                  if (((ds[dim].size == 1) and ((dim is not "t")
```
Comment (willirath) [inline xorca/lib.py]: These parens are very hard to parse.

What about:

```python
    to_squeeze = [dim for dim in ds.dims
                  if ((ds[dim].size == 1)
                      and ((dim is not "t") or
                           np.invert(np.issubdtype(ds[dim].dtype,
                                                   np.datetime64)))
                      and (dim not in z_dims))]
```

Comment (willirath) [reply xorca/lib.py]: Or, more verbose but self-explaining (check flake for the white space between funcs):

```python
    def _is_singleton(ds, dim):
        return (ds[dim] == 1)

    def _is_time_dim(ds, dim):
        return (dim in orca_names.t_dims or
                np.invert(np.issubdtype(ds[dim].dtype,
                                        np.datetime64)))

    def _is_z_dim(ds, dim):
        return (dim in orca_names.z_dims)

    to_squeeze = [dim for dim in ds.dims
                  if (not _is_singleton(ds, dim) and
                      not _is_time_dim(ds, dim) and
                      not _is_z_dim(ds, dim))]

**Comment by willirath** [comment]
```
E       AttributeError: module 'xorca.orca_names' has no attribute 't_dims'
```

**Comment by willirath** [comment]
Travis errors read like there is some non-scalar object returned by the new functions we defined.

**Comment by willirath** [comment]
Thanks @jk-rieck !
